pkg/k8s/watchers/watcher.go

```diff
@@ -536,7 +537,10 @@ func (k *K8sWatcher) enableK8sWatchers(ctx context.Context, resourceNames []stri
 
 func (k *K8sWatcher) k8sServiceHandler() {
 	eventHandler := func(event k8s.ServiceEvent) {
-		defer event.SWG.Done()
+		defer func(startTime time.Time) {
+			event.SWG.Done()
+			k.K8sServiceEventProcessed(event.Action.String(), event.Service.String(), startTime)
```

Comment:
The cardinality that a `event.Service.String()` has is too high for a production metric. /cc @marseel

Reply:
Hey @aanm,
We can safely remove the metric label for service if you think the cardinality is too high. It will be already tagged per namespaces anyway so I'm happy to remove the service label. What you think?

Reply:
> Hey @aanm, We can safely remove the metric label for service if you think the cardinality is too high. The metric will be already tagged per namespaces anyway so I'm happy to remove the service label. What you think?

Which namespace?

Reply:
Ah sorry, the metric namespace will always be reported as `kube-system` since the metric is published by cilium-agent. Anyway, I think we can safely remove the service label as the same code is invoked for each particular service.

Reply:
> Ah sorry, the metric namespace will always be reported as `kube-system` since the metric is published by cilium-agent. Anyway, I think we can safely remove the service label as the same code is invoked for each particular service.

Sorry, I don't understand, what's the proposed solution?

Reply:
```suggestion
			k.K8sServiceEventProcessed(event.Action.String(), startTime)
```

Reply:
@aanm I've updated the branch with all the changes I'm proposing to reduce the cardinality.

Reply:
It makes more sense to me now. I'll leave the final review to Marcel to take a 2nd look.